backend/core/src/secondary-adapters/postgres/user/user-repository-adapter.ts

```diff
@@ -11,6 +12,31 @@ export class UserRepositoryAdapter implements UserRepository {
   constructor(@inject(CONTAINER_TYPES.PrismaClient) prismaClient: PrismaClient) {
     this.prismaClient = prismaClient;
   }
+  async getPreferedCategories({ id }: DefaultRequestParam): Promise<UserGetPreferencesDto | null> {
+    return this.prismaClient.user.findFirst({
+      where: {
+        id,
+      },
+      select: {
+        videoPreferences: true,
+      },
+    });
+  }
+  async bindCategories({ id, categories }: UserBindCategoriesDto): Promise<Category[]> {
+    const values = categories.map((categoryId) => {
+      return `('${categoryId}', '${id}')`;
+    });
+    const query = `insert into "_CategoryToUser" ("A", "B") values ${values.join(',')} on conflict do nothing`;
```

Comment:
I can see SQL injection coming. I suggest you use parametrized queries  or `queryRaw` (not unsafe) https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access